examples/infer/parallel-prefix.desc

```diff
@@ -0,0 +1,247 @@
+                        let mut not_done = true;
+                        let mut endIndex = curr_block_id_x - 5;
+                        if endIndex < 0 {
+                            endIndex = 0
+                        };
+                        // TODO implement !done
+                        while not_done {
+                            *t_exclusive_prefix = 0;
+                            let mut i = 1;
+                            let mut flag = 0;
+                            let mut agg = 0;
+                            let mut prefix = 0;
+                            let mut not_break_loop = true;
+                            while i <= 5 && curr_block_id_x - i >= 0 && not_break_loop {
+                                let raw_ptr_flag = to_raw_ptr(&uniq *t_flag);
+                                let raw_ptr_flag_offset = offset_raw_ptr::<i32>(raw_ptr_flag, 0 - i);
+
+                                let raw_ptr_agg = to_raw_ptr(&uniq *t_agg);
+                                let raw_ptr_agg_offset = offset_raw_ptr::<i32>(raw_ptr_agg, 0 - i);
+
+                                let raw_ptr_prefix = to_raw_ptr(&uniq *t_prefix);
+                                let raw_ptr_prefix_offset = offset_raw_ptr::<i32>(raw_ptr_prefix, 0 - i);
+
+                                unsafe {
+                                    flag = *raw_ptr_flag_offset;
+                                    agg = *raw_ptr_agg_offset;
+                                    prefix = *raw_ptr_prefix_offset
+                                };
+
+                                if flag == 0 {
+                                    not_break_loop = false
+                                };
+                                if flag == 1 {
+                                    *t_exclusive_prefix = *t_exclusive_prefix + agg
+                                };
+                                if flag == 2 {
+                                    *t_exclusive_prefix = *t_exclusive_prefix + prefix;
+                                    not_break_loop = false;
+                                    not_done = false
+                                };
+                                i = i - 1
+                            }
+                        }
+                    }
+                };
+                parfor thread in split_thread_grp::<1>(block).0
+                with t_r, t_prefix, t_exclusive_prefix from &uniq *result_view, &uniq *b_prefix, &shrd *view_exclusive_prefix  {
+                    *t_prefix = *t_exclusive_prefix + *t_r
+                };
+                // Cheesy Threadfence
+                parfor thread in split_thread_grp::<1>(block).0
+                with t_flag from &uniq *b_flag {
+                    *t_flag = 2
+                };
+
+                // upsweep::<b_size, t_size>(block, &uniq shared_input)
+                // upsweep(block, &uniq shared_input);
+                {
+                    let foo = &uniq shared_input;
```

Comment:
Need to use foo because if the code would be following:
```rust
                    // let foo = &uniq shared_input;
                    for_nat d in halved_range(b_size) <'a>{
                        let tmp_up_view = group_mut::<(b_size * t_size)/d, 'a, gpu.shared, (b_size * t_size), i32>(
                            to_view_mut(&uniq *shared_input));
```

The transpilation fails with

```
thread 'main' panicked at 'Is the type dead? `Data(DataTy { dty: At(DataTy { dty: Array(DataTy { dty: Scalar(I32), constraints: [], span: None }, BinOp(Mul, Ident(Ident { name: "b_size", span: Some(Span { begin: 2122, end: 2128 }), is_implicit: false }), Ident(Ident { name: "t_size", span: Some(Span { begin: 2131, end: 2137 }), is_implicit: false }))), constraints: [], span: None }, GpuShared), constraints: [], span: None })`', src/ty_check/borrow_check.rs:75:18
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: descend::ty_check::borrow_check::ownership_safe
             at ./src/ty_check/borrow_check.rs:75:18
             ....
```